sdk/identity/azure-identity/azure/identity/aio/_credentials/managed_identity.py

```diff
@@ -58,7 +56,9 @@ def __init__(self, **kwargs: "Any") -> None:
         :rtype: :class:`azure.core.credentials.AccessToken`
         :raises ~azure.identity.CredentialUnavailableError: managed identity isn't available in the hosting environment
         """
-        return AccessToken()
+        if not self._credential:
+            raise CredentialUnavailableError(message="No managed identity endpoints found")
```

Comment:
This is nitpicky and probably no one would ever notice 😊 but: in current implementations there is only ever one endpoint, so we should write "endpoint" here